File: spmercantil/src/main/java/github/faustofjunqueira/spmercantil/core/dto/Page.java
package github.faustofjunqueira.spmercantil.core.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.List;

/**
 * Representação de registros paginado da base de dados
 *
 * @param <TData> Tipo do registro carregado
 */
@Data
@AllArgsConstructor
public class Page<TData> {

    /**
     * total de registro presentes na base de dados
     */
    private Long total;

    /**
     *  lista registro da pagina
     */
    private List<TData> data;

}

File: SpMercantil/Core/Domain/Dto/Page.cs
using System.Collections.Generic;

namespace Core.Domain.Dto
{
    /// <summary>
    ///     Representação de registros paginado da base de dados
    /// </summary>
    /// <typeparam name="TData">Tipo do registro carregado</typeparam>
    public class Page<TData>
    {
        /// <summary>
        ///     total de registro presentes na base de dados
        /// </summary>
        public long Total { get; set; }

        /// <summary>
        ///     lista registro da pagina
        /// </summary>
        public List<TData> Data { get; set; } = new List<TData>();
    }
}